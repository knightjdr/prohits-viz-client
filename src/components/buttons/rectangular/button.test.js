import React from 'react';
import { render } from '@testing-library/react';

import Button from './button';

const theme = {
  fontDark: '#111',
};

const renderElement = props => render(<Button {...props} />);

describe('Rectangular button style', () => {
  const props = { theme };

  it('should match snapshot', () => {
    const { container } = renderElement(props);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should set color on focus', () => {
    const { container } = renderElement(props);
    const modifier = {
      modifier: '&:focus:not([disabled])',
    };
    expect(container.firstChild).toHaveStyleRule('color', theme.fontDark, modifier);
  });

  it('should set color on hover', () => {
    const { container } = renderElement(props);
    const modifier = {
      modifier: '&:hover:not([disabled])',
    };
    expect(container.firstChild).toHaveStyleRule('color', theme.fontDark, modifier);
  });
});
