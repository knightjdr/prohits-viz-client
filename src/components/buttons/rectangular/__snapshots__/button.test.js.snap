// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rectangular button style should match snapshot 1`] = `
.c0 {
  background-color: transparent;
  border-color: transparent;
  box-sizing: border-box;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: background-color,border-color,color;
  transition-property: background-color,border-color,color;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0:focus:not([disabled]),
.c0:hover:not([disabled]) {
  background-color: #fff;
}

.c0:active {
  -webkit-animation-duration: 0.3s;
  animation-duration: 0.3s;
  -webkit-animation-name: click;
  animation-name: click;
}

.c1 {
  border-radius: 3px;
  border-style: solid;
  border-width: 2px;
  box-sizing: border-box;
  font-size: inherit;
  padding: 0.2em 0.5em;
}

.c1:focus:not([disabled]),
.c1:hover:not([disabled]) {
  color: #111;
}

<button
  class="c0 c1"
  kind="primary"
/>
`;
